backend/open_api_spec.yml

```diff
@@ -0,0 +1,829 @@
+
+  /curation/auth/token:
+    post:
+      summary: Generate an access token
+      security:
+        - apiKey: []
+      responses:
+        "201":
+          description: Created
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  access_token:
+                    type: string
+        "401":
+          $ref: "#/components/responses/401"
+
+  /curation/collections:
+    get:
+      description: Fetch all Collections metadata. Authentication required for private collections.
+      security:
+        - cxguserCookieLenient: []
+      parameters:
+        - $ref: "#/components/parameters/query_visibility"
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  collections:
+                    type: object
+                    properties:
+                      public:
+                        type: array
+                        items:
+                          $ref: "#/components/schemas/collection_with_dataset_preview"
+                      private:
+                        type: array
+                        items:
+                          $ref: "#/components/schemas/collection_with_dataset_preview"
+        "401":
+          $ref: "#/components/responses/401"
+    post:
+      description: Create a new Collection
+      security:
+        - cxguserCookie: []
+      requestBody:
+        content:
+          application/json:
+            $ref: "#/components/schemas/collection_metadata"
+      responses:
+        "201":
+          description: Created
+            content:
+              application/json:
+                schema:
+                  $ref: "#/components/schemas/collection_uuid"
+        "401":
+          $ref: "#/components/responses/401"
+
+  /curation/collections/{collection_id}:
+    delete:
+      description: Delete a private Collection. Delete an existing revision for a public Collection.
+      security:
+        - cxguserCookie: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              type: string
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+    get:
+      description: Fetch Collection metadata and associated Datasets, public and private.
+      security:
+        - cxgUserCookieLenient: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+        - $ref: "#/components/parameters/query_visibility"
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              $ref: "#/components/schemas/collection"
```

Comment:
I think here would be a good place to return both the public and the private in the same response. Similar to how you wanted to return the GET /collections endpoint.

Reply:
Since it's only two, sure. I agree!